src/lib/enquirer.js

```diff
@@ -0,0 +1,58 @@
+const Enquirer = require('enquirer');
+const chalk = require('chalk');
+
+function disablePromptsInCI(answers = {}) {
+  return enquirer => {
+    const prompt = enquirer.prompt.bind(enquirer);
+    const context = { ...enquirer.answers, ...answers };
+
+    // eslint-disable-next-line no-param-reassign
+    enquirer.prompt = async questions => {
+      const list = [].concat(questions || []);
+
+      // Purposefully loading `ci-info` here so we can adequately test it.
+      const ci = require('ci-info'); // eslint-disable-line global-require
+      if (!ci.isCI) {
+        // If we aren't in a CI environment then we shouldn't look for unanswered questions.
+        return prompt(list);
+      }
+
+      const missingArgs = [];
+
+      (questions || []).forEach(item => {
+        // Confirm-style questions are always optional!
+        if (item.type !== 'confirm') {
+          const value = context[item.name];
+          if (value === undefined && !item.skip()) {
+            missingArgs.push(item.name);
+          }
+        }
+      });
+
+      if (missingArgs.length) {
+        throw new Error(
+          `We've detected you're running within a ${
+            ci.name
+          } environment, please supply the following required arguments for your command instead of relying on our terminal prompt system: ${chalk.yellow(
+            missingArgs.join(',')
+          )}`
```

Comment:
This might be too casual (and I'm not sure if prettier will like this), so zhuzh this up as you see fit! I also removed the `chalk` usage:
```suggestion
          `Oops! We noticed you're running this command within a ${
            ci.name
          } environment, but we have a few questions we were going to ask you! We'll need the following arguments for your command to run question-free: ${
            missingArgs.join(',')
          }`
```